spatial/r3/mat.go

```diff
@@ -187,12 +185,12 @@ func (m *Mat) Outer(alpha float64, x, y Vec) {
 	m.Set(2, 2, az*y.Z)
 }
 
-// Sets the rotation component of this matrix to the rotation specified by q,
-// as outlined at https://en.wikipedia.org/wiki/Rotation_matrix#Quaternion.
+// RotationFromQuat converts the quaternion q to the corresponding orthogonal
+// matrix and stores the result in the receiver. q should be normalized but this
```

Comment:
Why not normalize the quaternion inside the function?

Reply:
In the case that the user want a rotation and scale, that becomes impossible.

Reply:
But then we're in an undocumented territory and the existing documentation and naming becomes inaccurate: the matrix is not orthogonal and so not a rotation.

Reply:
Fair enough; probably we should do that then and note that it's normalised. Thanks for that.

Reply:
Yeah, I think if scaling is needed it can be easily achieved with `Scale`.

Reply:
I've been thinking about this a bit more and it's probably the cleanest to leave it as is. The behavior is documented for unit quaternions, non-unit behavior is not documented and we don't need to test it. We could maybe replace "should" with "must", @kortschak ?

Reply:
That's fine with me. I think that people should be using quaternions for this kind of thing where all this just works.